api/pom.xml

```diff
@@ -141,6 +141,10 @@
          <groupId>org.apache.lucene</groupId>
          <artifactId>lucene-queries</artifactId>
       </dependency>
+	   <dependency>
+		   <groupId>org.apache.lucene</groupId>
+		   <artifactId>lucene-backward-codecs</artifactId>
+	   </dependency>
```

Comment:
Reverted.

Reply:
Hey there, two of my colleagues ran a few tests and rediscovered why this is needed. When e.g. initialising OpenMRS with release 2.2.x and then updating with TRUNK-4830-N the following exception is raised:

`Caused by: java.lang.IllegalArgumentException: An SPI class of type org.apache.lucene.codecs.Codec with name 'Lucene410' does not exist.  You need to add the corresponding JAR file supporting this SPI to your classpath.  The current classpath supports the following names: [Lucene54]`

This is resolved by adding the backwards codecs.

Reply:
Added the dependency and re-run the very last integration test on my side...

Reply:
Re-resolving as the test went well...

Reply:
That's interesting. I guess we need to ensure the Lucene index is updated earlier in the update process. The index gets rebuilt at the end of the process. I'll open a separate ticket for that.

Reply:
And that now exists as [TRUNK-5731](https://issues.openmrs.org/browse/TRUNK-5731).

Reply:
Why not just delete the lucene folder in the application data directory?

Reply:
I had a quick look into the code base, there is a property `hibernate.search.default.indexBase=%APPLICATION_DATA_DIRECTORY%/lucene/indexes` but it does not seem to be used by the core codebase so there is not easy fix. Instead more tests need to be written around that before changing it.

My recommendation would be to deal with the lifecycle of the search index in a separate ticket. This ticket happened to highlight the issue because of a rare test (initialise with one version, update with a more recent version). It is good to have noticed that but fixing this is not in scope of Liquibase snapshots.

What do you think?

Reply:
I completely agree. 👍

Reply:
@WolfSchlegel do we still need lucene-backward-codecs?